### PR TITLE
Importing command from text

### PR DESCRIPTION
**Pull request by piterm11**

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/command/file/IImportCommand.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package edu.kis.powp.jobs2d.command.file;
+
+import edu.kis.powp.jobs2d.command.DriverCommand;
+
+import java.util.List;
+
+public interface IImportCommand {
+
+	List<DriverCommand> importCommandSequence(String filename);
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/file/IImportCommand.java]: importing should be done from a text, not from a file. Reading text from a file is a separate task (responsibility)

File: src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java [hunk 2]
```diff
@@ -6,11 +6,16 @@
...
+import javax.swing.*;
+import javax.swing.filechooser.FileSystemView;
 
 import edu.kis.powp.appbase.gui.WindowComponent;
+import edu.kis.powp.jobs2d.command.ComplexCommand;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.file.IImportCommand;
+import edu.kis.powp.jobs2d.command.file.ImportCMDFileCommand;
+import edu.kis.powp.jobs2d.command.file.ImportJSONFileCommand;
+import edu.kis.powp.jobs2d.command.file.ImportXMLFileCommand;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java]: Try to remove these dependencies. An appropriate importer could be supplied from an external facility (e.g., kind of a factory).

File: src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java [hunk 3]
```diff
@@ -6,11 +6,16 @@
...
 import java.util.List;
 
-import javax.swing.JButton;
-import javax.swing.JFrame;
-import javax.swing.JTextArea;
+import javax.swing.*;
+import javax.swing.filechooser.FileSystemView;
 
 import edu.kis.powp.appbase.gui.WindowComponent;
+import edu.kis.powp.jobs2d.command.ComplexCommand;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java]: Is it used?!

**Review by kowallus — CHANGES_REQUESTED** [review]
Follow suggestions please.

**Comment by piterm11** [comment]
Fixed

File: src/main/java/edu/kis/powp/jobs2d/command/file/ImportCMDFileCommand.java [hunk 4]
```diff
@@ -8,19 +8,20 @@
 import java.io.FileReader;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/file/ImportCMDFileCommand.java]: Remove unused imports

File: src/main/java/edu/kis/powp/jobs2d/command/file/IImportCommand.java [hunk 5]
```diff
@@ -7,4 +7,8 @@
 public interface IImportCommand {
 
 	List<DriverCommand> importCommandSequence(String filename);
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/file/IImportCommand.java]: Please update the name of the param

File: src/main/java/edu/kis/powp/jobs2d/command/file/ImportJSONFileCommand.java [hunk 6]
```diff
@@ -17,17 +17,12 @@
...
 		List<DriverCommand> commandList = new ArrayList<>();
 		JSONParser jsonParser = new JSONParser();
 		try {
-			FileReader fileReader = new FileReader(filename);
-			JSONArray commandArray = (JSONArray) jsonParser.parse(fileReader);
+			JSONArray commandArray = (JSONArray) jsonParser.parse(text);
 			commandArray.forEach(jo -> commandList.add(getCommand((JSONObject) jo)));
-		} catch (FileNotFoundException e) {
-			e.printStackTrace();
-		} catch (IOException e) {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/file/ImportJSONFileCommand.java]: Cleanup imports

File: src/main/java/edu/kis/powp/jobs2d/command/file/ImportXMLFileCommand.java [hunk 7]
```diff
@@ -10,21 +11,21 @@
...
-import java.io.File;
 import java.io.IOException;
 
 public class ImportXMLFileCommand implements IImportCommand {
 	private final List<DriverCommand> commandList = new ArrayList<>();
 
 	@Override
 	public List<DriverCommand> importCommandSequence(String filename) {
-		processFile(filename);
+		processText(filename);
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/file/ImportXMLFileCommand.java]: Only a comment: this private method seems unnecessary.

File: src/main/java/edu/kis/powp/jobs2d/command/file/ImporterFactory.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+	public static IImportCommand getImporter(Type importerType) {
+		switch (importerType) {
+			case CMD:
+				return new ImportCMDFileCommand();
+			case JSON:
+				return new ImportJSONFileCommand();
+			case XML:
+				return new ImportXMLFileCommand();
+			default:
+				throw new IllegalArgumentException();
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/file/ImporterFactory.java]: Is this handled gracefully?

File: src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java [hunk 9]
```diff
@@ -90,26 +91,28 @@ private void clearCommand() {
...
-		else if (path.endsWith(".cmd")) {
-			importCommand = new ImportCMDFileCommand();
-		}
-		else return;
-		List<DriverCommand> commandList = importCommand.importCommandSequence(path);
+		String[] elements = path.split("\\.");
+		String extension = elements[elements.length - 1].toUpperCase();
+		IImportCommand importCommand = ImporterFactory.getImporter(IImportCommand.Type.valueOf(extension));
+		String text = getTextFromFile(path);
+		List<DriverCommand> commandList = importCommand.importCommandSequence(text);
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java]: Only checking: how do you handle invalid input?

File: src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java [hunk 10]
```diff
@@ -4,18 +4,19 @@
...
 import javax.swing.filechooser.FileSystemView;
 
 import edu.kis.powp.appbase.gui.WindowComponent;
-import edu.kis.powp.jobs2d.command.ComplexCommand;
 import edu.kis.powp.jobs2d.command.DriverCommand;
-import edu.kis.powp.jobs2d.command.file.IImportCommand;
-import edu.kis.powp.jobs2d.command.file.ImportCMDFileCommand;
-import edu.kis.powp.jobs2d.command.file.ImportJSONFileCommand;
-import edu.kis.powp.jobs2d.command.file.ImportXMLFileCommand;
+import edu.kis.powp.jobs2d.command.file.*;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java]: why *? How many imports from file package you use?

**Review by kowallus — CHANGES_REQUESTED** [review]
Please apply minor cleanup

**Comment by pioter00** [comment]
fixed

**Pull request merged**